src/ZPublisher/HTTPRequest.py

```diff
@@ -1204,11 +1204,11 @@ def resolve_url(self, url):
         # method is called on the response).
         try:
             object = req.traverse(path)
-        except Exception:
+        except Exception as exc:
             rsp.exception()
-        if object is None:
-            req.clear()
-            raise sys.exc_info()[0](rsp.errmsg)
+            if object is None:
+                req.clear()
+                raise exc.__class__(rsp.errmsg)
```

Comment:
That's true. That check was outside the except block, but assumes an exception was raised. In Python 3 sys.exc_info was (None, None, None) there, so I moved the check inside the except block and use the exc directly. The object is None check can go than.